### PR TITLE
Add `threading._profile_hook` to `threading.pyi`

### PR DESCRIPTION
**Pull request by eggplants**
Ref: #7424

### CODE REVIEW

File: stdlib/threading.pyi [hunk 1]
```diff
@@ -8,6 +8,8 @@ _TF = Callable[[FrameType, str, Any], Optional[Callable[..., Any]]]
 _PF = Callable[[FrameType, str, Any], None]
 _T = TypeVar("_T")
 
+_profile_hook = Optional[_TF]
```
Comment (AlexWaygood) [inline stdlib/threading.pyi:11]:
```suggestion
_profile_hook: _TF | None
```

**Review by hauntsaninja — COMMENTED** [review]
This should match the type of `setprofile`, right? So `_PF`?

Also nit: might be nice to move this below the `__all__` definition

**Comment by eggplants** [comment]
> This should match the type of `setprofile`, right? So `_PF`?

You're right, sorry for my mistake...

**Review by hauntsaninja — APPROVED** [review]
Looks great, thanks!

**Pull request merged**